source/declarations.tex

```diff
@@ -2935,8 +2935,12 @@
   using C::x;
   int x(double);
 };
-int f(D* d) {
-  return d->x();    // ambiguous: \tcode{B::x} or \tcode{C::x}
+void f(D* d) {
+   d->x();       // ambiguous: \tcode{x}'s direct parent \tcode{A} is an ambiguous base of the naming class \tcode{D}
```

Comment:
We never use the word "parent" in the context of classes. (We only use it in the context of threads and directories.) "Parent-child" is a *terrible* metaphor for class inheritance. A better metaphor is "base-derived".